feat: add transition effect to mobile navbar

rishvant opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow our contribution guidelines
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**

- Improved mobile navbar with smooth transitions.
- Adjusted z-index values to ensure proper layering and fix overlapping issues.
- Enhanced overall user experience with better visibility and smoother animations.

**Related issue(s)**
#625 
<!-- If you refer to a particular issue, provide its number, otherwise, remove this section.
For example, `Resolves #123`, `Fixes #43`, or `See also #33`. The `See also #33` option will not automatically close the issue after the PR merge. -->

AceTheCreator reviewed (changes requested):
This looks good! I noticed a black border bottom line in the background, which I believe shouldn’t be there. Also, the spacing between the first nav item and the logo is too much.

rishvant commented:
@AceTheCreator I have fixed as per your suggestions, now have a look at that.

ashmit-coder commented:
Hey @rishvant i noticed that when we click on the venue button in the navbar, we are taken to the top of the page in the background.


https://github.com/user-attachments/assets/88695679-f4fb-4545-9386-f75adb73acad

rishvant commented:
@ashmit-coder Its a bug and its present in deployed website as well, should I create a new issue for this or fix it in this only?

rishvant commented:
@ashmit-coder @AceTheCreator I have fixed the issue, please check if its fine

AceTheCreator reviewed (changes requested):
Looks good @rishvant 

Could you maybe add some little changes? 

- In the current venue dropdown, i will really appreciate if you can fix the location line break, which is happening basically due to container width
- No indication on the venue mobile nav that indicates it a dropdown

rishvant commented:
@AceTheCreator According to your suggestion, I have fixed those changes:

- Added nowrap property to remove line break in venue dropdown
- Added a dropdown icon in mobile nav after venue

AceTheCreator commented:
/rtm
